### PR TITLE
feat: shared configuration

### PR DESCRIPTION
**Pull request by mpetrun5**
Implement fetching shared configuration from network storage.

## Related Issue Or Context

Closes: #68 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -140,29 +145,43 @@ func Run() error {
...
 				eventHandlers = append(eventHandlers, listener.NewRefreshEventHandler(topologyProvider, topologyStore, tssListener, coordinator, host, communication, connectionGate, keyshareStore, bridgeAddress))
 				eventHandlers = append(eventHandlers, listener.NewRetryEventHandler(tssListener, depositHandler, bridgeAddress, *config.GeneralChainConfig.Id, config.BlockConfirmations))
-				evmListener := coreListener.NewEVMListener(client, eventHandlers, blockstore, config)
-
-				mh := coreExecutor.NewEVMMessageHandler(bridgeContract)
-				mh.RegisterMessageHandler(config.Erc20Handler, coreExecutor.ERC20MessageHandler)
-				mh.RegisterMessageHandler(config.Erc721Handler, coreExecutor.ERC721MessageHandler)
-				mh.RegisterMessageHandler(config.GenericHandler, coreExecutor.GenericMessageHandler)
-				mh.RegisterMessageHandler(pGenericHandler, executor.PermissionlessGenericMessageHandler)
+				evmListener := coreListener.NewEVMListener(client, eventHandlers, blockstore, *config.GeneralChainConfig.Id, config.BlockRetryInterval, config.BlockConfirmations, config.BlockInterval)
```
Comment (P1sar) [inline app/app.go:181]: I am wondering why do we have *uint8 type of ID in core library, looks wrong to me...

File: config/config.go [hunk 2]
```diff
@@ -36,56 +40,82 @@ type RawConfig struct {
...
 	}
 
-	return processRawConfig(rawConfig)
+	return processRawConfig(rawConfig, config)
 }
 
-func processRawConfig(rawConfig RawConfig) (Config, error) {
-	config := Config{}
+// GetConfigFromNetwork fetches shared configuration from URL and parses it.
+func GetConfigFromNetwork(url string, config *Config) (*Config, error) {
```
Comment (P1sar) [inline config/config.go:74]: I would rename it then accordingly to something like GetChainConfigFromNetwork or GetSharedConfigFromNetwork

File: app/app.go (hunk 1 above)
Comment (mpetrun5) [reply app/app.go:181]: Its uint8 on solidity as well. Why it is a pointer I have no idea and it is before my time I think.

Comment (mpetrun5) [reply app/app.go:181]: It's not really a problem IMO and bit of a hassle as we'll have to fix in core so I wouldn't deal with this now.

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**